extension/oauth2clientauthextension/extension_test.go

```diff
@@ -272,3 +276,44 @@ func TestOAuthExtensionShutdown(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Nil(t, oAuthExtensionAuth.Shutdown(context.Background()))
 }
+
+func TestFailContactingOAuth(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(200)
+		w.Write([]byte("not-json"))
+	}))
+	defer server.Close()
+
+	serverURL, err := url.Parse(server.URL)
+	assert.NoError(t, err)
+
+	oauth2Authenticator, err := newClientCredentialsExtension(&Config{
+		ClientID:     "dummy",
+		ClientSecret: "ABC",
+		TokenURL:     serverURL.String(),
+	}, zap.NewNop())
+	assert.Nil(t, err)
+
+	// Test for gRPC connections
+	credential, err := oauth2Authenticator.PerRPCCredentials()
+	assert.Nil(t, err)
+
+	_, err = credential.GetRequestMetadata(context.Background())
+	assert.ErrorIs(t, err, ErrFailedToGetSecurityToken)
+	assert.Contains(t, err.Error(), "failed to get security token from token endpoint")
+
+	// Test for HTTP connections
+	setting := confighttp.HTTPClientSettings{
+		Endpoint: "http://example.com/",
+		CustomRoundTripper: func(next http.RoundTripper) (http.RoundTripper, error) {
+			return oauth2Authenticator.RoundTripper(next)
+		},
+	}
+
+	client, _ := setting.ToClient(componenttest.NewNopHost().GetExtensions())
+	req, err := http.NewRequest("POST", setting.Endpoint, nil)
+	assert.NoError(t, err)
+	_, err = client.Do(req)
+	assert.ErrorIs(t, err, ErrFailedToGetSecurityToken)
+	assert.Contains(t, err.Error(), "failed to get security token from token endpoint")
```

Comment:
Probably better test will be to test if the err.Error() has the token url rather than string inside ErrFailedToGetSecurityToken as we are already testing this in previous line
```suggestion
	assert.Contains(t, err.Error(), serverURL.String())
```